mobile/src/test/java/ch/epfl/sdp/mobile/application/chess/parser/StringCombinatorsTest.kt

```diff
@@ -0,0 +1,29 @@
+package ch.epfl.sdp.mobile.application.chess.parser
+
+import ch.epfl.sdp.mobile.application.chess.parser.StringCombinators.string
+import org.junit.Assert.*
+import org.junit.Test
+
+class StringCombinatorsTest {
+
+  @Test
+  fun given_aString_when_parsed_then_returnSetOfResult() {
+    val res = string().parse("This is a test")
+    val expected = Parser.Result("is a test", "This")
+    assertEquals(expected, res.first())
+  }
+
+  @Test
+  fun given_aString_when_parsedWithAGiven1stWord_then_returnSetOfResult() {
+    val res = string("This").parse("This is a test")
+    val expected = Parser.Result("is a test", "This")
+    assertEquals(expected, res.first())
+  }
+
+  @Test
+  fun given_aEmptyString_when_parsed_then_returnEmptySet() {
+    val res = string().parse("")
+    val expected = emptySequence<Parser<String, Token>>()
+    assertEquals(expected, res)
+  }
```

Comment:
I think it would be interesting to have a test where you try to parse a certain token, but it is not at the start of the string.
Something along the lines of 
```
val res = string("That").parse("This is a test")
    assertThat(res).isEmpty()
```

Reply:
Yeah, I added in the commit https://github.com/epfl-SDP/android/pull/280/commits/717fcfeced6ffd7d6f639ed1e8ad8e5dceb9aff8